core/sql/optimizer/BindItemExpr.cpp

```diff
@@ -10851,6 +10851,96 @@ ItemExpr *ZZZBinderFunction::bindNode(BindWA *bindWA)
       }
     break;
 
+    case ITM_CSV_FORMAT:
+      {
+	bindChildren(bindWA);
+	if (bindWA->errStatus()) 
+	  return this;
+
+        // The way the arguments of CSV_FORMAT are represented in
+        // the parse tree is as a tree of ItemList nodes; so
+        // CSV_FORMAT(a,b,c,d) is represented as
+        //
+        //    this
+        //    /  \
+        //   a   ItemList
+        //         /  \
+        //        b   ItemList
+        //              /  \
+        //             c    d
+        //
+        // The code below traverses accordingly.
+       
+        ItemExpr * next = child(0)->castToItemExpr();     
+        ItemExpr * resultBoundTree = NULL;
+        while (next)  // while arguments remain to process
+          {
+            ItemExpr * childk = NULL;
+            if (next->getOperatorType() == ITM_ITEM_LIST)
+              {
+                childk = next->child(0);
+                next = next->child(1);
+              }
+            else
+              {
+                childk = next;
+                next = NULL;
+              }
+           
+            const NAType &type = childk->getValueId().getType();
+            switch (type.getTypeQualifier())
+              {
+                case NA_BOOLEAN_TYPE:
+                case NA_DATETIME_TYPE:
+                case NA_INTERVAL_TYPE:
+                case NA_NUMERIC_TYPE:
+                  {
+                    // TODO: Is VARCHAR(100) big enough? Consider bignums, for example. We could
+                    // probably be smarter about the length. E.g. VARCHAR(6) is big enough for SMALLINT.
+                    strcpy(buf,"CAST(@A1 AS VARCHAR(100))");
+                    parseTree = parser.getItemExprTree(buf, strlen(buf), BINDITEMEXPR_STMTCHARSET, 1, childk);
+                    boundTree = parseTree->bindNode(bindWA);
+                    if (bindWA->errStatus()) 
+                      return this;
+                    break;
+                  }
+                case NA_CHARACTER_TYPE:
+                  {
+                    boundTree = childk;
+                    break;
+                  }
+                default:
+                  {
+                    // operand has an unsupported data type
+                    *CmpCommon::diags() << DgSqlCode(-4018)
+                                        << DgString0("CSV_FORMAT")
+                                        << DgString1(type.getTypeName().data());
+                    bindWA->setErrStatus();
+                    return this;
+                  }
+              }
+
+            strcpy(buf,"CASE WHEN POSITION(',' IN @A1) > 0 THEN '\"' || @A1 || '\"' ELSE @A1 END");
```

Comment:
One more thought: I think that in a CSV file, white space between fields is ignored. So, it may be best if this code adds a TRIM, so that the padding around fixed chars is not included.